common/src/main/java/org/apache/celeborn/common/network/server/memory/MemoryManager.java

```diff
@@ -69,37 +71,27 @@ public class MemoryManager {
   private boolean underPressure;
   private final AtomicBoolean trimInProcess = new AtomicBoolean(false);
 
-  // For buffer stream
+  // For credit stream
   private final AtomicLong readBufferCounter = new AtomicLong(0);
   private long readBufferThreshold = 0;
-  private final ReadBufferDispatcher readBufferDispatcher;
+  private long readBufferTarget = 0;
+  private ReadBufferDispatcher readBufferDispatcher;
+  private List<ReadBufferTargetChangeListener> readBufferTargetChangeListeners;
+  private Object readBufferTargetChangeLock = new Object();
```

Comment:
readBufferTargetChangeLock is unnecessary, just lock on readBufferTargetChangeListeners